CMakeLists.txt

```diff
@@ -51,6 +51,14 @@ endif()
 set(NVIM_VERSION_MAJOR 0)
 set(NVIM_VERSION_MINOR 1)
 set(NVIM_VERSION_PATCH 1)
+# Package maintainers should define this according to their package system
+# conventions, if the source code or runtime files were modified, e.g.:
+#   cmake -DNVIM_VERSION_PRERELEASE=-deb3
+# or define it as empty to prevent "-dev", if no modifications were made:
+#   cmake -DNVIM_VERSION_PRERELEASE=
+if(NOT DEFINED NVIM_VERSION_PRERELEASE)
+  set(NVIM_VERSION_PRERELEASE "-dev")
+endif()
```

Comment:
> I don't like this too much.. I think the version should not depend on how Nvim is built. If a source snapshot is not a release, it should have "-dev", otherwise it shouldn't. I agree that it's cumbersome to remove NVIM_VERSION_PRERELEASE when we tag a release, though.. but I don't have a better idea.

Another idea--that might be slightly better--is to include an extra file in the release, say .version-info.  We can then have CMake pickup on this file and coerce the version number (leaving off the "-dev", for instance), and when it's not there keep the current behavior.

We use a similar mechanism at our workplace so that we can control the version number via our QuickBuild instance, and it allows us to promote the builds through to a final release--ultimately dropping things out of the version that mark it as a dev or a release candidate.  We also use it to label our dev versions separately than the git commit numbers--because sometimes there are packaging issues and you need to rebuild from the same commit number, but it's not the same build.

The downside is that there's an extra file you need to include as part of the tag, but at least you don't have to remember to pass yet another argument.